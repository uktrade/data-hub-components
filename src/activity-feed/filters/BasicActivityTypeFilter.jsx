import React from 'react'
import styled from 'styled-components'
import {
  MEDIA_QUERIES,
  SPACING,
  BORDER_WIDTH_MOBILE,
} from '@govuk-react/constants'
import { GREY_2, GREY_3 } from 'govuk-colours'
import Checkbox from '@govuk-react/checkbox'
import PropTypes from 'prop-types'
import SelectFilter from './SelectFilter'

const StyledBasicFiltersContainer = styled('div')`
  padding: ${SPACING.SCALE_2};
  background: ${GREY_3};
`

const StyledActivityFeedFilterTitle = styled('div')`
  padding: 0;
  margin: 0;
  width: auto;
  display: block;

  ${MEDIA_QUERIES.DESKTOP} {
    display: inline-block;
  }
`

const StyledTitle = styled('h4')`
  margin: ${SPACING.SCALE_2} ${SPACING.SCALE_4} ${SPACING.SCALE_2} 0;
`

const StyledHr = styled('hr')`
  margin: ${SPACING.SCALE_2} 0;
  border-style: double;
  border-color: ${GREY_2};
`

const StyledCheckbox = styled(Checkbox)`
  min-height: 26px;
  margin-bottom: 0;

  & > span {
    margin: 0;
    padding: 0;

    &::before {
      background: white;
      width: 26px;
      height: 26px;
    }

    &::after {
      top: 6px;
      left: 5px;
      width: 12px;
      height: 6px;
      border-width: 0 0 ${BORDER_WIDTH_MOBILE} ${BORDER_WIDTH_MOBILE};
    }
  }

  input {
    width: 26px;
    height: 26px;
    margin: 0;
    padding: 0;
  }
`

export default class BasicActivityTypeFilter extends React.PureComponent {
  static propTypes = {
    activityTypeFilters: PropTypes.array.isRequired,
    filteredActivity: PropTypes.array.isRequired,
    onActivityTypeFilterChange: PropTypes.func.isRequired,
    onShowDetailsClick: PropTypes.func.isRequired,
    showDetails: PropTypes.bool.isRequired,
  }

  render() {
    const {
      activityTypeFilters,
      filteredActivity,
      onActivityTypeFilterChange,
      onShowDetailsClick,
      showDetails,
    } = this.props

    return (
      <StyledBasicFiltersContainer>
        <StyledActivityFeedFilterTitle>
          <StyledTitle>Filter by</StyledTitle>
        </StyledActivityFeedFilterTitle>
        <SelectFilter
          filters={activityTypeFilters}
          onActivityTypeFilterChange={onActivityTypeFilterChange}
          value={filteredActivity}
        />
        <StyledHr />
        <StyledCheckbox onChange={onShowDetailsClick} checked={showDetails}>
          Show details for all activities
        </StyledCheckbox>
      </StyledBasicFiltersContainer>
    )
  }
}
