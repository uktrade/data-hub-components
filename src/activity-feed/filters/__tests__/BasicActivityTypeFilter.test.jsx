import React from 'react'
import renderer from 'react-test-renderer'

import BasicActivityTypeFilter from '../BasicActivityTypeFilter'

describe('BasicActivityTypeFilter', () => {
  describe('when the details for all activities are hidden', () => {
    test('renders filters', () => {
      const tree = renderer
        .create(
          <BasicActivityTypeFilter
            activityTypeFilters={[]}
            filteredActivity={[]}
            onActivityTypeFilterChange={() => {}}
            onShowDetailsClick={() => {}}
            showDetails={false}
            value={[]}
          />
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
