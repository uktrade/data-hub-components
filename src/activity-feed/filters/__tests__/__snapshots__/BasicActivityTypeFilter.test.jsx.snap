// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicActivityTypeFilter when the details for all activities are hidden renders filters 1`] = `
.c9 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  position: relative;
  min-height: 40px;
  padding: 0 0 0 40px;
  clear: left;
  margin-bottom: 10px;
}

.c10 {
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  opacity: 0;
  cursor: pointer;
}

.c10:checked + span:after {
  opacity: 1;
}

.c10:focus + span:before {
  outline: 3px solid transparent;
  outline-offset: 3px;
  box-shadow: 0 0 0 3px #ffbf47;
}

.c10 + span {
  opacity: 1;
  pointer-events: auto;
}

.c11 {
  display: inline-block;
  cursor: pointer;
  padding: 8px 15px 5px;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
}

.c11::before {
  content: '';
  box-sizing: border-box;
  position: absolute;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  border: 2px solid black;
  background: transparent;
}

.c11::after {
  content: '';
  position: absolute;
  top: 11px;
  left: 9px;
  width: 18px;
  height: 7px;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
  border: solid;
  border-width: 0 0 5px 5px;
  border-top-color: transparent;
  background: transparent;
  opacity: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c4:after {
  content: '';
  display: table;
  clear: both;
}

.c5 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c6 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 33px;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
}

.c6:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
}

.c3 {
  background: #f8f8f8;
  padding: 5px;
  display: block;
}

.c3 label {
  display: block;
}

.c3 label span {
  display: inline-block;
  margin: 0 10px;
  width: auto;
  margin: 0;
}

.c3 label select {
  width: 100%;
}

.c0 {
  padding: 10px;
  background: #dee0e2;
}

.c1 {
  padding: 0;
  margin: 0;
  width: auto;
  display: block;
}

.c2 {
  margin: 10px 20px 10px 0;
}

.c7 {
  margin: 10px 0;
  border-style: double;
  border-color: #bfc1c3;
}

.c8 {
  min-height: 26px;
  margin-bottom: 0;
}

.c8 > span {
  margin: 0;
  padding: 0;
}

.c8 > span::before {
  background: white;
  width: 26px;
  height: 26px;
}

.c8 > span::after {
  top: 6px;
  left: 5px;
  width: 12px;
  height: 6px;
  border-width: 0 0 4px 4px;
}

.c8 input {
  width: 26px;
  height: 26px;
  margin: 0;
  padding: 0;
}

@media print {
  .c9 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c9 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c9 {
    margin-bottom: 10px;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    margin-bottom: 0;
  }
}

@media print {
  .c5 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c5 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    margin-bottom: 0;
  }
}

@media print {
  .c6 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    width: 50%;
    height: 38px;
  }
}

@media only screen and (min-width:769px) {
  .c3 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
  }
}

@media only screen and (min-width:769px) {
  .c3 label {
    display: inline-block;
  }
}

@media only screen and (min-width:769px) {
  .c3 label span {
    margin: 0 10px;
  }
}

@media only screen and (min-width:769px) {
  .c3 label select {
    width: auto;
  }
}

@media only screen and (min-width:769px) {
  .c1 {
    display: inline-block;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h4
      className="c2"
    >
      Filter by
    </h4>
  </div>
  <div
    className="c3"
  >
    <label
      className="c4"
      defaultValue="all"
      name="activity-types-filter"
      onChange={[Function]}
      value={Array []}
    >
      <span
        className="c5"
      >
        Activity types
      </span>
      <select
        className="c6"
      />
    </label>
  </div>
  <hr
    className="c7"
  />
  <label
    className="c8 c9"
  >
    <input
      checked={false}
      className="c10"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="c11"
    >
      Show details for all activities
    </span>
  </label>
</div>
`;
